rmw_freertps_cpp/CMakeLists.txt

```diff
@@ -45,6 +44,48 @@ register_rmw_implementation("rosidl_typesupport_freertps_cpp")
 if(AMENT_ENABLE_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
+
+  set(message_files
+    "msg/BoundedArrayBounded.msg"
+    "msg/BoundedArrayStatic.msg"
+    "msg/BoundedArrayUnbounded.msg"
+
+    "msg/Empty.msg"
+
+    "msg/LargeArray.msg"
+
+    "msg/PrimitiveStaticArrays.msg"
+
+    "msg/PrimitivesBounded.msg"
+    "msg/PrimitivesStatic.msg"
+    "msg/PrimitivesUnbounded.msg"
+
+    "msg/StaticArrayBounded.msg"
+    "msg/StaticArrayStatic.msg"
+    "msg/StaticArrayUnbounded.msg"
+
+    "msg/UnboundedArrayBounded.msg"
+    "msg/UnboundedArrayStatic.msg"
+    "msg/UnboundedArrayUnbounded.msg"
+  )
+
+  rosidl_generate_interfaces(${PROJECT_NAME}_msgs ${message_files}
+    SKIP_INSTALL
+  )
+
+  if(NOT WIN32)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra")
+  endif()
+
+  ament_add_gtest(test_serialization_cpp test/test_serialization.cpp)
+  if(TARGET test_serialization_cpp)
+    target_link_libraries(test_serialization_cpp
+      ${PROJECT_NAME}_msgs__rosidl_typesupport_freertps_cpp
+      ${_AMENT_EXPORT_ABSOLUTE_LIBRARIES}
+      ${_AMENT_EXPORT_LIBRARY_TARGETS})
+  endif()
```

Comment:
This test seems to not link against `rmw_freertps_cpp`. Why is it in this package then? Could it be moved to `rosidl_typesupport_freertps_cpp` instead?